base_layer/wallet/src/transaction_service/service.rs

```diff
@@ -673,6 +694,143 @@ where
         Ok(tx_id)
     }
 
+    /// Sends a one side payment transaction to a recipient
+    /// # Arguments
+    /// 'dest_pubkey': The Comms pubkey of the recipient node
+    /// 'amount': The amount of Tari to send to the recipient
+    /// 'fee_per_gram': The amount of fee per transaction gram to be included in transaction
+    pub async fn send_one_sided_transaction(
+        &mut self,
+        dest_pubkey: CommsPublicKey,
+        amount: MicroTari,
+        fee_per_gram: MicroTari,
+        message: String,
+        transaction_broadcast_join_handles: &mut FuturesUnordered<
+            JoinHandle<Result<u64, TransactionServiceProtocolError>>,
+        >,
+    ) -> Result<TxId, TransactionServiceError>
+    {
+        if self.node_identity.public_key() == &dest_pubkey {
+            warn!(target: LOG_TARGET, "One-sided spend-to-self transactions not supported");
+            return Err(TransactionServiceError::OneSidedTransactionError(
+                "One-sided spend-to-self transactions not supported".to_string(),
+            ));
+        }
+
+        // Prepare sender part of the transaction
+
+        let mut stp = self
+            .output_manager_service
+            .prepare_transaction_to_send(
+                amount,
+                fee_per_gram,
+                None,
+                message.clone(),
+                script!(PushPubKey(Box::new(dest_pubkey.clone()))),
```

Comment:
This script is not correct. 
It will verify for any 0 input. 

If we want p2pkh we need: 
`script!(Dup HashBlake256 PushHash(dest_pubkey.hash()) EqualVerify);`

if we want p2pk
`script!(Dup PushPubKey(dest_pubkey) EqualVerify);`

Reply:
I could easily change this, but first please confirm that the [RFC-0202](https://rfc.tari.com/RFC-0202_TariScriptOpcodes.html#one-sided-transactions) is wrong? It states:
_One-sided transactions lock the input to a predetermined public key provided by the recipient; essentially the same method that Bitcoin uses. The simplest form of this is to simply post the new owner's public key as the script: `PushPubkey(P_B)`_

Reply:
The simple script works as only the owner can produce the script signature. Do we really need to also have them confirm that they know the destination public key in the P2PKH example? p2pk adds no value as far as I can tell.

Reply:
You are both correct, I stand corrected. `PushPubkey(P_B)` is in all essence the same as p2pk. 

Anyone can provide a valid input to this script as it's an empty stack. But only the true recipient can sign the script offset as p_B is required for that and only the owner of P_B should know that.